test/libsolidity/syntaxTests/pragma/broken_version_1.sol

```diff
@@ -0,0 +1,3 @@
+pragma solidity ^0^1;
```

Comment:
Why is this parsed? And into what exactly?

Reply:
Dunno

Reply:
Apparently it only parsed until `^0`, validating that major is 0, and ignored the rest. Fixed.

Reply:
Apparently the tests we have suggest there's some weird support for these. Will split the PR into to.